src/commands/email_layout/list.ts

```diff
@@ -0,0 +1,107 @@
+import { Flags, ux } from "@oclif/core";
+import { AxiosResponse } from "axios";
+
+import * as ApiV1 from "@/lib/api-v1";
+import BaseCommand from "@/lib/base-command";
+import { formatDate } from "@/lib/helpers/date";
+import { merge } from "@/lib/helpers/object";
+import {
+  maybePromptPageAction,
+  pageFlags,
+  paramsForPageAction,
+} from "@/lib/helpers/page";
+import { withSpinner } from "@/lib/helpers/request";
+import * as EmailLayout from "@/lib/marshal/email_layout";
+
+export default class EmailLayoutList extends BaseCommand<
+  typeof EmailLayoutList
+> {
+  static summary = "Display all email layouts for an environment.";
+
+  static flags = {
+    environment: Flags.string({
+      default: "development",
+      summary: "The environment to use.",
+    }),
+    "hide-uncommitted-changes": Flags.boolean({
+      summary: "Hide any uncommitted changes.",
+    }),
+    ...pageFlags,
+  };
+
+  static enableJsonFlag = true;
+
+  async run(): Promise<ApiV1.ListEmailLayoutResp | void> {
+    const resp = await this.request();
+
+    const { flags } = this.props;
+    if (flags.json) return resp.data;
+
+    this.render(resp.data);
+  }
+
+  async request(
+    pageParams = {},
+  ): Promise<AxiosResponse<ApiV1.ListEmailLayoutResp>> {
+    const props = merge(this.props, { flags: { ...pageParams } });
+
+    return withSpinner<ApiV1.ListEmailLayoutResp>(() =>
+      this.apiV1.listEmailLayouts(props),
+    );
+  }
+
+  async render(data: ApiV1.ListEmailLayoutResp): Promise<void> {
+    const { entries } = data;
+    const { environment: env, "hide-uncommitted-changes": commitedOnly } =
+      this.props.flags;
+
+    const qualifier =
+      env === "development" && !commitedOnly ? "(including uncommitted)" : "";
+
+    this.log(
+      `‣ Showing ${entries.length} email layouts in \`${env}\` environment ${qualifier}\n`,
+    );
+
+    /*
+     * Email layouts list table
+     */
+
+    ux.table(entries, {
```

Comment:
I decided not to show the `html_layout` and `text_layout`, as the data here is too big to fit in the table.

Also the `footer_links` are being parsed to show up to 3 links `keys` + the count of the rest of links.

I first though of showing something like `link` - `value`, but that doesn't fit in the screen properly.

What do you think?

<img width="895" alt="Screen Shot 2023-09-21 at 11 13 25" src="https://github.com/knocklabs/knock-cli/assets/67347884/b8d61e36-771a-4915-bd26-9cf94d14e1a6">

Reply:
> I decided not to show the html_layout and text_layout, as the data here is too big to fit in the table.

Yep that makes sense.

> Also the footer_links are being parsed to show up to 3 links keys + the count of the rest of links.
> What do you think?

This could work but I'm actually kinda leaning toward not having the footer links column in the list view. The way I go about formatting these cli command screens is usually by mirroring the dashboard equivalents. And looking at the list page in dashboard ([example](https://dashboard.knock.app/vidyard/development/layouts)), we just show the key and the name plus timestamps. 

So that would be my vote, but let me know if you disagree!

Reply:
Got it I see your point and I agree, here are the changes: https://github.com/knocklabs/knock-cli/pull/229/commits/57d56260c81fa46dc24378433d8d6b1ec10e7ee1